src/utils/pages/pages.ts

```diff
@@ -11,21 +11,24 @@ export type Pages<T> = {
 
 export function createPages<T>(resourceResponse: ResourceResponse<T>, originalParams: RequestParams<T>): Pages<T> {
   let pages: Pages<T> = {
-    prev: new RequestParams<T>().clone(originalParams),
-    current: new RequestParams<T>().clone(originalParams),
-    next: new RequestParams<T>().clone(originalParams),
+    prev: { ...originalParams },
```

Comment:
That still copies the properties of the original ones by reference.
E.g.
```
let foo = {bar: {baz: 1}};
let qux = {...foo};
qux.bar.baz = 2;
console.log(foo.bar.baz); // 2
```
A simple, but not so pretty, way to address it would be `JSON.parse(JSON.stringify(originalParam))`.
Or we could grab lodash and go with `cloneDeep` :smile:

Reply:
Should be fine as `RequestParams` is `Record<string, string>` and therefore cannot contain object references